src/frontends/tensorflow_common/src/op/truncate_mod.cpp

```diff
@@ -0,0 +1,28 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "common_op_table.hpp"
+#include "openvino/op/floor_mod.hpp"
+
+using namespace std;
+using namespace ov::opset10;
+
+namespace ov {
+namespace frontend {
+namespace tensorflow {
+namespace op {
+OutputVector translate_truncate_mod_op(const NodeContext& node) {
+    default_op_checks(node, 2, {"TruncateMod"});
+    auto x = node.get_input(0);
+    auto y = node.get_input(1);
+
+    auto trunc_mod = make_shared<FloorMod>(x, y);
```

Comment:
`FloorMod` is not quite correct because it applies Python-style truncation (zero-away rounding for negative division).
`TruncateMod` required C-truncation. So for this you need have a sub-graph: 1) compute `FloorMod` 2) compute a mask where just one of operands is negative 3) negate result of `FloorMod` for positive mask.

Examples: TrucateMod(7, 5) = 2; TruncateMod(8, 4) = 0; Truncate(-7, 5) = -2; TruncateMod(-8, 4) = 0

Reply:
Agree, thanks